src-plugins/mscAlgorithmPaint/mscAlgorithmPaintToolBox.h

```diff
@@ -95,7 +95,9 @@ public slots:
     void activateMagicWand();
     void updateMagicWandComputationSpeed();
 
-    void copyMetaDataToPaintedData();
+    void copyMetaData(medAbstractData* output,
+                      medAbstractData* input);
+    dtkSmartPointer<medAbstractData> deepCopy(medAbstractData* input);
```

Comment:
You should replace ```deepCopy``` with an overload of ```dtkAbstractObject::clone()``` in ```itkDataImage```. (See the comments in dtkAbstractObject.cpp.).